init.lua

```diff
@@ -140,6 +148,17 @@ local function node_is_water_source(node)
 	return (node.name == "default:water_source")
 end
 
+local function node_is_river_water_source(node)
+	if not node then
+		return false
+	end
+	if node.name == "ignore" then
+		return false
+	end
+
+	return (node.name == "default:river_water_source")
```

Comment:
You could move this to a constant, such as `NODE_RIVER_WATER_SOURCE` and do an in-line comparison (without this function) where needed.